src/cluster_legacy.c

```diff
@@ -5318,6 +5321,8 @@ void clusterSetMaster(clusterNode *n) {
         myself->flags &= ~(CLUSTER_NODE_MASTER|CLUSTER_NODE_MIGRATE_TO);
         myself->flags |= CLUSTER_NODE_SLAVE;
         clusterCloseAllSlots();
+        /* Cancel all ASM tasks when switching into slave */
+        clusterAsmCancel(NULL, "switching to replica");
```

Comment:
Hi @tezc  not sure if the cluster plugin has  this function to allow us to cancel asm tasks

if not, we can check the change of role in asmBeforeSleep, or other places, but it looks like ugly.

Reply:
no, i mean do we have a similar `clusterSetMaster` function, switching a master to replica

Reply:
I remember we talked about this before, plugin would call us to notify on config change but not yet implemented. Perhaps, plugin may cancel externally but we can also internally cancel it on this notification (just to avoid ugly scenarios in case there is a bug in plugin)

Reply:
make sense